Upgrade frameworks and tooling

jploskey opened the pull request:
Note: Github Workflows have still not been ported over to the new tooling (using `uv`). That will need to be done eventually, but I don't have time to do that before my PTO starting November 5th so maybe we can do another PR after this one. However, all the frontend is working and so is the new Python virtual environment, at least for downloading charts.

**Major Changes**

Frontend:

1. Switch to Vue 3.
2. Change the build system from Vue CLI/Webpack to Nuxt/Vite. This allows automatic routing, and far easier project management.
3. Using [Nuxt](https://nuxt.com) for server side rendering. This will make it easy to get proper SEO working when we'd like to make the site public.
4. Switch from Vue Bootstrap to [NuxtUI](https://ui.nuxt.com).
5. Use TypeScript for type checking.
6. Use latest ESLint and Prettier for linting and formatting.
7. Switch from `yarn` to `npm` for project management. The features of Yarn have largely been integrated into NPM and it's rarely used on new projects anymore. For a faster package manager that's compatible, you can use `pnpm`.

Python:

1. Switch from using `miniconda` and `pip` to [`uv`](https://docs.astral.sh/uv/). Trust me on this. `uv` is the best package manager I've ever used, and it will manage (downloading and installing) Python versions for projects automatically.
2. Require Python 3.13. No need to install this yourself, `uv` will handle it automatically.
3. NOTE: The CI workflows have NOT yet been updated. I'll get to this after coming back from vacation Thursday next week.

**Other Changes**

Frontend:

1. The minimum required NodeJS version has likely increased due to the requirements of dependencies. So if running `npm install` fails with a NodeJS version error, I recommend installing the latest Node LTS version.
4. A few visual differences, including the sliders and tabs looking different due to using a different UI framework. I've also moved the sliders in plot viewer components to the bottom, which felt more intuitive. Willing to move them back if desired.
5. Most CSS defined in templates has been removed and replaced with [TailwindCSS](https://tailwindcss.com) classes. The NuxtUI framework uses Tailwind internally so this is the easiest way to go. I'd never used Tailwind before, but it's very easy to learn.
6. Script commands in `package.json` have changed slightly. This includes `serve` being renamed to `dev`, and `preview` being renamed to `serve`.

**Testing**

1. Clone this repository into a new directory, separate from your existing development directory and switch to this branch. (`feature/framework-upgrade`)
2. Install `uv`: https://docs.astral.sh/uv/getting-started/installation/. It should only require running a single command shell command. I don't recommend installing using `brew` or other system-level package managers as `uv` will manage its own versioning.
3. Enter the cloned project directory and run `uv sync`. This will download all required packages, install a compatible Python version, and create a virtual environment at the root of the project in `.venv`. 
4. Activate the virtual environment using `source .venv/bin/activate` if on a Unix system, otherwise if on Windows, run `.\.venv\scripts\activate`.
5. Download plots using `python dev/setup_dev_images.py`.
6. Enter the `dashboard` directory.
7. Run `nom install` to install frontend packages. If it says your Node version is too old, install the latest LTS version.
8. Run `npm run dev` to run the project using a development server with hot-reloading, or `npm run build` then `npm run serve` to run a production build.

jdduprey commented:
@jploskey This looks great. All the views/features I've tested so far seem identical but slightly better looking and more responsive. I'm going to download some SVGs to make sure the sorting is unchanged there as well. 

Could you update the README with your workflow for getting a dev environment set up with npm? and remove now outdated steps like installing  Vue CLI/Webpack. I managed to get everything working with your notes in the PR, but would be good to keeps the README up to date.

jdduprey commented:
We can actually remove all python dependencies except `git+https://github.com/OOI-CabledArray/rca-data-tools.git@main` because rca-data-tools has them all as dependencies itself. We could also get rid of environment.yaml in the project root all together,  and just use uv/pyproject.toml.

jploskey commented:
@jdduprey Thanks for the review! I've updated the main `README.md` and removed those unnecessary dependencies from `pyproject.toml`. I'm going to leave the `environment.yaml` alone for now, as all those CI workflows depend on it. I'm thinking we can address that and remove Conda fully in another PR.

jdduprey commented:
There is some missing css for SVG objects, which is my fault since I did not include them in the s3 bucket download script.

jdduprey commented on dashboard/app/components/plots/Overview.vue:
```diff
@@ -0,0 +1,305 @@
+            <img v-if="isPNG(url)" class="h-auto max-w-full" loading="lazy" :src="url" />
+            <object
+              v-if="isSVG(url)"
+              class="h-auto max-w-[1500px] w-100"
```
I think this class has to be changed to make the svgs match the png css.

jploskey replied:
Ah, `w-100` should be `w-full`.

jdduprey commented:
This is ready to merge when Jake is back. The only remaining thing is to update the deployment workflow and decide if we want to switch to server-side rendering now or at some point down the road.
